### ISSUE TITLE
check: pull trust anchor certificate from `linkerd-identity-trust-roots`

### ISSUE THREAD

**Issue by mateiidavid** [linkerd/linkerd2#7449]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What is the issue?

We introduced `linkerd-identity-trust-roots` as a configmap to help people have more control when it comes to managing trust anchors for their linkerd installation.

It was pointed out that when installing linkerd with `externalCA` and `externalIssuer` options, the check will fail when matching the trust bundle the proxies have with what's stored in linkerd's config. This is also the case when using extensions that have their own checks, e.g `viz`.

```
Cannot connect to Linkerd Viz: Some pods do not have the current trust bundle and must be restarted:
        * grafana-6fbc6dbf86-mz282
        * metrics-api-545b8d457b-jzdq7
        * prometheus-5d747596d-hfdlb
        * tap-85f7b8b69c-z6jxn
        * tap-injector-6fc7ddd46c-68pc6
        * web-75447677d7-cdl2z
Validate the install with: linkerd viz check
```

I think the problem comes from the fact that during installation, linkerd will store the trust root in both `linkerd-config` and `linkerd-identity-trust-roots`. When doing the check, the healthchecker will use the trust from `linkerd-config`. If linkerd is  installed without manually passing in the certificates, then there will be a mismatch.

### How can it be reproduced?

Install linkerd with a CA managed by cert-manager, or with an existing `linkerd-identity-trust-roots`.

### Logs, error output, etc

× viz extension proxies are healthy
    Some pods do not have the current trust bundle and must be restarted:
        * grafana-6fbc6dbf86-mz282
        * metrics-api-545b8d457b-jzdq7
        * prometheus-5d747596d-hfdlb
        * tap-85f7b8b69c-z6jxn
        * tap-injector-6fc7ddd46c-68pc6
        * web-75447677d7-cdl2z
    see https://linkerd.io/2.11/checks/#l5d-viz-proxy-healthy for hints

### output of `linkerd check -o short`

```text
N/A
```

### Environment

N/A

### Possible solution

In the healthchecker, fetch the certificates from the trust-root configmap.

**Comment by wim-de-groot** [comment]
It seems in [checkPodsProxiesCertificate](https://github.com/linkerd/linkerd2/blob/main/pkg/healthcheck/healthcheck.go#L2167) that the trust anchors of the pods are compared to the `linkerd-config` instead of with the `linkerd-identity-trust-roots`.

**Comment by wim-de-groot** [comment]
The config-map in kubernetes looks like this:
```yaml
identity:
      externalCA: true
      issuer:
        clockSkewAllowance: 20s
        externalCA: false
        issuanceLifetime: 24h0m0s
        scheme: kubernetes.io/tls
        tls:
          crtPEM: ""
```

But in the [Identity](https://github.com/linkerd/linkerd2/blob/main/pkg/charts/linkerd2/values.go#L189) struct the value `externalCA` is not shown. There is also another boolean shown in the Identity struct that is not shown in the `linker-config` called `ServiceAccountTokenProjection`. I searched the code for places where `Identity.ExternalCA` is used but I only found [identity.yaml](https://github.com/linkerd/linkerd2/blob/main/charts/linkerd-control-plane/templates/identity.yaml#L22).

**Comment by mateiidavid** [comment]
@wim-de-groot the fix sounds good to me, but I think we can simplify this a bit futher. We can change the healthchecker to just read the trust root (`CA`) directly from `linkerd-identity-trust-roots` instead of `linkerd-config`, without any additional logic around the `externalCA` flag.

We use `linkerd-config` mainly to store the install-time values when linkerd is installed through the cli. I think since the change was originally implemented (see https://github.com/linkerd/linkerd2/issues/6452) we decided to have the trust roots configmap as the source of truth. `linkerd-config` is at the moment used (_I think_) only for installs and upgrades as a way of storing the configuration; `linkerd-identity-trust-roots` is actually mounted in the proxies and in the proxy-injector to bootstrap identity for all pods. I opened a separate issue to document this a little bit better.

Even if the CA is not external, it'll still be stored in `linkerd-identity-trust-roots`, that's why we can avoid checking the `externalCA` flag and simply read from the CA configmap in all check cases.

Let me know if this makes sense. We'd also appreciate a contribution here, so if it all sounds good and you're willing to get a PR in, we'd be more than happy to guide you through it. :)

**Comment by wim-de-groot** [comment]
Yes this makes a lot of sense, but I didn't know that it was a possibility. I thought at first that the `linkerd-identity-trust-roots` config map was only created when the `externalCA` was enabled.

**Comment by wim-de-groot** [comment]
@mateiidavid I would love to get a PR in. Already updated the method and the associated test. So what do I need to do?

**Comment by mateiidavid** [comment]
> Yes this makes a lot of sense, but I didn't know that it was a possibility. I thought at first that the `linkerd-identity-trust-roots` config map was only created when the `externalCA` was enabled. This makes fixing this issue so much easier.

Yeah, this is actually a bit tricky to understand at first. If we can do anything to improve docs let us know. For reference, if you need to revisit the subject down the line, https://github.com/linkerd/linkerd2/issues/6452 explains the change (and why it was needed).

> @mateiidavid I would love to get a PR in. Already updated the method and the associated test. So what do I need to do?

Awesome! Thanks for taking this on! 🥳 

 Just fork the project and open a PR against our main branch and from there, we'll review and discuss the change a bit more in the PR if we have any comments :)
